daemon/cmd/hubble.go

```diff
@@ -153,61 +153,63 @@ func (d *Daemon) launchHubble() {
+		localSrvOpts := []serveroption.Option{
+			serveroption.WithUnixSocketListener(sockPath),
+			serveroption.WithHealthService(),
+			serveroption.WithObserverService(d.hubbleObserver),
+			serveroption.WithPeerService(peerSvc),
+			serveroption.WithInsecure(),
+		}
 
-	localSrvOpts := []serveroption.Option{
-		serveroption.WithUnixSocketListener(sockPath),
-		serveroption.WithHealthService(),
-		serveroption.WithObserverService(d.hubbleObserver),
-		serveroption.WithPeerService(peer.NewService(d.nodeDiscovery.Manager, peerServiceOptions...)),
-		serveroption.WithInsecure(),
-	}
+		localSrvOpts = d.addRecorderServiceOpt(logger, localSrvOpts)
 
-	if option.Config.EnableRecorder && option.Config.EnableHubbleRecorderAPI {
-		dispatch, err := sink.NewDispatch(option.Config.HubbleRecorderSinkQueueSize)
+		localSrv, err := server.NewServer(logger, localSrvOpts...)
 		if err != nil {
-			logger.WithError(err).Error("Failed to initialize Hubble recorder sink dispatch")
+			logger.WithError(err).Error("Failed to initialize local Hubble server")
 			return
 		}
-		d.monitorAgent.RegisterNewConsumer(dispatch)
-		svc, err := recorder.NewService(d.rec, dispatch,
-			recorderoption.WithStoragePath(option.Config.HubbleRecorderStoragePath))
-		if err != nil {
-			logger.WithError(err).Error("Failed to initialize Hubble recorder service")
+		logger.WithField("address", sockPath).Info("Starting local Hubble server")
+		if err := localSrv.Serve(); err != nil {
+			logger.WithError(err).Error("Failed to start local Hubble server")
 			return
 		}
-		localSrvOpts = append(localSrvOpts, serveroption.WithRecorderService(svc))
-	}
-
-	localSrv, err := server.NewServer(logger, localSrvOpts...)
-	if err != nil {
-		logger.WithError(err).Error("Failed to initialize local Hubble server")
-		return
-	}
-	logger.WithField("address", sockPath).Info("Starting local Hubble server")
-	if err := localSrv.Serve(); err != nil {
-		logger.WithError(err).Error("Failed to start local Hubble server")
-		return
+		go func() {
+			<-d.ctx.Done()
+			localSrv.Stop()
+		}()
+	} else {
+		logger.Info("Local Hubble Server was not enabled")
 	}
-	go func() {
-		<-d.ctx.Done()
-		localSrv.Stop()
-	}()
 
 	// configure another hubble instance that serve fewer gRPC services
 	address := option.Config.HubbleListenAddress
 	if address != "" {
 		if option.Config.HubbleTLSDisabled {
 			logger.WithField("address", address).Warn("Hubble server will be exposing its API insecurely on this address")
 		}
+		if peerSvc == nil {
+			var peerServiceOptions []serviceoption.Option
+			if option.Config.HubbleTLSDisabled {
+				peerServiceOptions = append(peerServiceOptions, serviceoption.WithoutTLSInfo())
+			}
+			peerSvc = peer.NewService(d.nodeDiscovery.Manager, peerServiceOptions...)
+		}
 		options := []serveroption.Option{
 			serveroption.WithTCPListener(address),
 			serveroption.WithHealthService(),
+			serveroption.WithPeerService(peerSvc),
```

Comment:
Unless I am missing something, this line addition is all we need on the hubble server side.